source/draft_api.rst

```diff
@@ -656,50 +716,37 @@ The following content is checked by the API for validity:
 * URI must be a valid URI.
 * Activities must exist in the database.
 * The Project must exist in the database.
+* Project slugs must not already belong to another project.
 * The owner of the request must be the user in the time submission.
     * This is authorization not authentication.
 
-.. note::
-
-    While they won't produce an error, empty data structures such as ``""`` and
-    ``[]`` will be ignored when sent to the api as the value of an object
-    variable.
-
-.. note::
-
-   When an object is updated it's ``parents`` is soft-deleted and a copy is
-   created with the new information. This results in the object having an
-   incremented 'revision' field the updated information specifed in the POST
-   request.
-
 
 ----------------
 
 DELETE Endpoints
 ----------------
 
-A DELETE request sent to any object's endpoint (e.g. */projects/<slug>*) will
-result in the deletion of the object from the records.
+The single object endpoints (e.g. ``/times/:uuid``, ``/projects/:slug``) support
+DELETE requests; these remove an object from the records.
 
-If the object exists and the ``?include_deleted=true`` parameter is not passed,
-the API will return a 200 OK status with an empty response body.
+These objects must always be soft-deleted; that is, the object will still exist within the
+database. Nonetheless, requests for lists of objects (e.g. ``GET /projects``) will exclude
+the object from the results, and requests for single objects (e.g.
+``GET /times/:uuid``) will return a 404. The parameter ``?include_deleted``
+circumvents this requirement and allows deleted objects to be returned as well.
 
-If the object does not exist or has ``deleted_at`` set to a non-null datetime
-and the ``?include_deleted=true`` parameter is not passed, the API will return
-a 404 Object Not Found error (see error docs) .
+An object's deleted status is indicated by setting its ``deleted_at`` to the time of
+deletion; if the value is null, the object is not deleted. Only the most recent revision
+is set. In addition, activities and projects have their ``slugs`` set to null, in order
```

Comment:
Yeah, because how you get rid of the slugs depends on your implementation. In SQL, we keep them in a separate table and DELETE them on this delete; in something like Mongo, you'd just set them to null. So this is a little more database agnostic.